ext/scimv2/logic/src/main/java/org/apache/syncope/core/logic/SCIMDataBinder.java

```diff
@@ -835,206 +1167,231 @@ protected void setAttribute(
-                setAttribute(userUR.getPlainAttrs(), conf.getUserConf().getDisplayName(), op);
+                case "displayName" ->
+                        setAttribute(userURs.get(numberUR.get()).getPlainAttrs(), conf.getUserConf().getDisplayName(),
+                                op);
 
-            case "nickName" ->
-                setAttribute(userUR.getPlainAttrs(), conf.getUserConf().getNickName(), op);
+                case "nickName" ->
+                        setAttribute(userURs.get(numberUR.get()).getPlainAttrs(), conf.getUserConf().getNickName(), op);
 
-            case "profileUrl" ->
-                setAttribute(userUR.getPlainAttrs(), conf.getUserConf().getProfileUrl(), op);
+                case "profileUrl" ->
+                        setAttribute(userURs.get(numberUR.get()).getPlainAttrs(), conf.getUserConf().getProfileUrl(),
+                                op);
 
-            case "title" ->
-                setAttribute(userUR.getPlainAttrs(), conf.getUserConf().getTitle(), op);
+                case "title" ->
+                        setAttribute(userURs.get(numberUR.get()).getPlainAttrs(), conf.getUserConf().getTitle(), op);
 
-            case "userType" ->
-                setAttribute(userUR.getPlainAttrs(), conf.getUserConf().getUserType(), op);
+                case "userType" ->
+                        setAttribute(userURs.get(numberUR.get()).getPlainAttrs(), conf.getUserConf().getUserType(), op);
 
-            case "preferredLanguage" ->
-                setAttribute(userUR.getPlainAttrs(), conf.getUserConf().getPreferredLanguage(), op);
+                case "preferredLanguage" -> setAttribute(userURs.get(numberUR.get()).getPlainAttrs(),
+                        conf.getUserConf().getPreferredLanguage(), op);
 
-            case "locale" ->
-                setAttribute(userUR.getPlainAttrs(), conf.getUserConf().getLocale(), op);
+                case "locale" ->
+                        setAttribute(userURs.get(numberUR.get()).getPlainAttrs(), conf.getUserConf().getLocale(), op);
 
-            case "timezone" ->
-                setAttribute(userUR.getPlainAttrs(), conf.getUserConf().getTimezone(), op);
+                case "timezone" ->
+                        setAttribute(userURs.get(numberUR.get()).getPlainAttrs(), conf.getUserConf().getTimezone(), op);
 
-            case "emails" -> {
-                if (!CollectionUtils.isEmpty(op.getValue()) && op.getValue().getFirst() instanceof SCIMUser) {
-                    setAttribute(userUR.getPlainAttrs(), conf.getUserConf().getEmails(),
-                            ((SCIMUser) op.getValue().getFirst()).getEmails(), op.getOp());
-                } else if (op.getPath().getFilter() != null) {
-                    setAttribute(userUR.getPlainAttrs(), conf.getUserConf().getEmails(), op);
+                case "emails" -> {
+                    if (!CollectionUtils.isEmpty(op.getValue()) && op.getValue().getFirst() instanceof SCIMUser) {
```

Comment:
Replace with

```
op.getValue().getFirst() instanceof SCIMUser scimUser
```

to avoid type cast

Please also check all other `instanceof` usages